### PR TITLE
[hue] Allow handling of QuantityType for color temperature channel

### PR DESCRIPTION
**Pull request by cweitkamp**
- Allow handling of QuantityType for color temperature channel

see https://github.com/openhab/openhab-core/pull/3129

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/LightStateConverter.java
Comment (jlaur) [inline bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/LightStateConverter.java:168]: Can you update Javadoc also here, since it's no longer the given `DecimalType`?

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks for the quick fix, only one very minor comment.

**Comment by lolodomo** [comment]
@jlaur : let me have a look before merging. There was an inappropriate log that should also be fixed.

**Review by jlaur — APPROVED** [review]
LGTM - @lolodomo, please merge when you have checked what you need to check.

**Comment by ccutrer** [comment]
Implementation LGTM. May want to update the README to note that the item type should be Number:Temperature now.

**Comment by jlaur** [comment]
> May want to update the README to note that the item type should be Number:Temperature now.

Oh, right - thanks for spotting this.

**Comment by cweitkamp** [comment]
I checked the README, fixed the logging messages and reduced log level.

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueGroupHandler.java
Comment (ccutrer) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueGroupHandler.java:235]: Yes it needs to be toInvertibleUnit. Mireds are technically of dimension 1/Temperature, so the dimension changes and you have to use the method that allows that to happen.  Good catch @lolodomo; I was paying more attention to if he got the command and the update in both handler types.

Comment (cweitkamp) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueGroupHandler.java]: Thanks for the hint. I changed it and added another unit test for sending Mired command to the channel.

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Comment by lolodomo** [comment]
Can you please submit the same change to the branch 3.4.x so that you can include it in a 3.4 patch ?

**Comment by lolodomo** [comment]
@cweitkamp : did you test the fix?

**Comment by cweitkamp** [comment]
We should cherry-pick such patches to the 3.4.0 branch and not create additional pull requests.

No, I did not test it in a real environment. I am sure it will work because of the unit tests.

**Comment by jlaur** [comment]
I have cherry-picked the commit into 3.4.x and built a JAR: [org.openhab.binding.hue-3.4.1-SNAPSHOT.jar](https://drive.google.com/file/d/1fHvwYTWWC3oQh90afaH8D8dVm0kbEJ7x/view?usp=sharing)

**Comment by lolodomo** [comment]
> I have cherry-picked the commit into 3.4.x

@jlaur : did you do that for all PRs marked with the "patch" label? 
You do that with Git or directly with GitHub?

**Comment by jlaur** [comment]
@lolodomo - yes, all PR's which are currently merged and labelled with "patch" have been cherry-picked. You can see them here: https://github.com/openhab/openhab-addons/commits/3.4.x

I used Git for that.

**Pull request merged**
